app/scripts/metamask-controller.js

```diff
@@ -2501,7 +2510,11 @@ export default class MetamaskController extends EventEmitter {
     if (!keyring.mnemonic) {
       throw new Error('Primary keyring mnemonic unavailable.');
     }
-    return keyring.mnemonic;
+
+    const recoveredIndices = Array.from(
+      new Uint16Array(new Uint8Array(keyring.mnemonic).buffer),
+    );
+    return recoveredIndices.map((i) => englishWordlist[i]).join(' ');
```

Comment:
`getPrimaryKeyringMnemonic` is used by Snaps which still passes mnemonics around as strings. They will be refactoring to pass mnemonics as `Uint8Arrays` but until then we must recover the string from the `Uint8Array` format that is now on the keyring from the KeyringController as of this latest version bump:
```
 const [keyring] = this.keyringController.getKeyringsByType(
      KEYRING_TYPES.HD_KEY_TREE,
    );
```